cTask.c

```diff
@@ -16,33 +16,48 @@ Please, take into account, that function must process full range INT value
 
 uint8_t *task1_intToStr(int32_t number)
 {
-    static uint8_t buffer[11];
-    for (int8_t i = 9; i>=0; i--)
-    {
-        buffer[i] = number % 10 + '0';
-        number /= 10;
-    }
-    for (int8_t i = 0; i<=9; i++)
-    {
-        if (buffer[0] == 0 + '0')
+    static uint8_t buffer[12];
+    uint8_t flag = 0;
```

Comment:
All over the code flag use as a begin index. "flag" - bad name for this variable